### PR TITLE
FAT-135 MODTAGS: Implement API Karate tests

### PR DESCRIPTION
**Pull request by Serhii-Zdorenko**
### **Purpose**
Karate implementation of existing Postman API tests for mod-tags

**Acceptance criteria:**
API Tests are created at https://github.com/folio-org/folio-integration-tests
Test run results are visible in TestRail: https://foliotest.testrail.io/index.php?/suites/overview/19
Automated regular runs are executed at https://jenkins-aws.indexdata.com/job/FOLIO_Reference_Builds/job/folio-integration/
Help team ramp on Karate
Help team estimate remaining Karate user stories in the backlog

### **Learning**
https://issues.folio.org/browse/FAT-135

### CODE REVIEW

**Review by psmagin — COMMENTED** [review]
Please add negative scenarios:
* when trying to create a new tag with label that already exists in the system;
* when trying to update a tag with a label that already exists in the system.

**Comment by Serhii-Zdorenko** [comment]
@psmagin Done

**Review by psmagin — APPROVED** [review]
Please, include ticket number to your commit messages in future.

File: mod-tags/src/test/resources/domain/tags/features/tags.feature [hunk 1]
```diff
@@ -6,57 +6,156 @@ Feature: Mod-tags integration tests
...
- # ================= positive test cases =================
-  @Undefined
+  @Positive
   Scenario: GET '/tags' should return 200 and collection of tags
-    * print 'undefined'
+    Given path 'tags'
+    When method GET
+    Then status 200
+    And match responseType == 'json'
+    And match response.totalRecords == 2
```
Comment (oleksandr-dekin) [inline mod-tags/src/test/resources/domain/tags/features/tags.feature:23]: You can add verification for label and description

File: mod-tags/src/test/resources/domain/tags/features/tags.feature [hunk 2]
```diff
@@ -6,57 +6,156 @@ Feature: Mod-tags integration tests
...
+    Given path 'tags/' + response.id
+    When method DELETE
+    And status 204
 
-  @Undefined
+  @Positive
+  Scenario: PUT '/tags/{id}' should return 204 if tag was successfully updated
+    Given path 'tags'
+    And request tag
+    When method POST
```
Comment (oleksandr-dekin) [inline mod-tags/src/test/resources/domain/tags/features/tags.feature:55]: You should verify status for every request. Please, add: `Then status 201`

File: mod-tags/src/test/resources/domain/tags/features/tags.feature [hunk 3]
```diff
@@ -6,57 +6,156 @@ Feature: Mod-tags integration tests
...
\ No newline at end of file
+    Given path 'tags'
+    And request invalid_tag
+    When method POST
+    Then status 422
+
+  @Negative
+  Scenario: POST '/tags' should return 422 when label is already exist
+    Given path 'tags'
+    When method GET
```
Comment (oleksandr-dekin) [inline mod-tags/src/test/resources/domain/tags/features/tags.feature:141]: The same, please, add verification status

File: mod-tags/src/test/resources/domain/tags/features/tags.feature [hunk 4]
```diff
@@ -6,57 +6,156 @@ Feature: Mod-tags integration tests
...
+    Then status 422
+
+  @Negative
+  Scenario: PUT '/tags' should return 422 when label is already exist
+    Given path 'tags'
+    When method GET
+    * def first_tag = response.tags[0]
+    * def second_tag = response.tags[1]
+
+    Given path 'tags/' + first_tag.id
```
Comment (oleksandr-dekin) [inline mod-tags/src/test/resources/domain/tags/features/tags.feature:157]: Please, avoid unnecessary character `/` 
```suggestion
    Given path 'tags', first_tag.id
```

File: mod-tags/src/test/resources/domain/tags/features/tags.feature [hunk 5]
```diff
@@ -6,57 +6,156 @@ Feature: Mod-tags integration tests
...
+    And status 204
+
+  @Positive
+  Scenario: DELETE '/tags/{id}' should return 204 if tag was successfully deleted
+    Given path 'tags'
+    And request tag
+    When method POST
+    And status 201
+
+    Given path 'tags/' + response.id
```
Comment (oleksandr-dekin) [inline mod-tags/src/test/resources/domain/tags/features/tags.feature:82]:
```suggestion
    Given path 'tags', response.id
```

File: mod-tags/src/test/resources/domain/tags/features/tags.feature [hunk 6]
```diff
@@ -6,57 +6,156 @@ Feature: Mod-tags integration tests
...
+    When method PUT
+    Then status 204
+
+    Given path 'tags/' + test_tag.id
+    When method GET
+    And response.id == test_tag.id
+    And response.label == test_tag.label
+    And response.description == test_tag.description
+
+    Given path 'tags/' + test_tag.id
```
Comment (oleksandr-dekin) [inline mod-tags/src/test/resources/domain/tags/features/tags.feature:71]: Given path 'tags/' + test_tag.id
```suggestion
    Given path 'tags', test_tag.id
```

File: mod-tags/src/test/resources/domain/tags/features/tags.feature [hunk 7]
```diff
@@ -6,57 +6,156 @@ Feature: Mod-tags integration tests
...
+    * def test_tag = response
+
+    Given path 'tags/' + test_tag.id
+    And request test_tag
+    And set test_tag.description = 'new description'
+    And set test_tag.label = 'new label'
+    When method PUT
+    Then status 204
+
+    Given path 'tags/' + test_tag.id
```
Comment (oleksandr-dekin) [inline mod-tags/src/test/resources/domain/tags/features/tags.feature:65]:
```suggestion
    Given path 'tags', test_tag.id
```

**Review by oleksandr-dekin — CHANGES_REQUESTED** [review]

**Review by oleksandr-dekin — APPROVED** [review]

**Pull request merged**